File: lightbulb-recycleradapters/src/main/java/com/github/RooneyAndShadows/lightbulb/recycleradapters/EasyAdapterDataModel.java
package com.github.RooneyAndShadows.lightbulb.recycleradapters;

import android.os.Parcel;
import android.os.Parcelable;

public abstract class EasyAdapterDataModel implements Parcelable {
    private boolean selected;

    public abstract String getItemName();

    public EasyAdapterDataModel(boolean selected) {
        this.selected = selected;
    }

    protected EasyAdapterDataModel(Parcel in) {
        selected = in.readByte() != 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeByte((byte) (selected ? 1 : 0));
    }

    public boolean isSelected() {
        return selected;
    }

    void setSelected(boolean selected) {
        this.selected = selected;
    }
}

File: lightbulb-recycleradapters/src/main/java/com/github/rooneyandshadows/lightbulb/recycleradapters/abstraction/data/EasyAdapterDataModel.kt
package com.github.rooneyandshadows.lightbulb.recycleradapters.abstraction.data

import android.os.Parcel
import android.os.Parcelable

abstract class EasyAdapterDataModel() : Parcelable {
    abstract val itemName: String

    override fun writeToParcel(p0: Parcel, p1: Int) {

    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<EasyAdapterDataModel> {
        override fun createFromParcel(parcel: Parcel): EasyAdapterDataModel? {
            return null
        }

        override fun newArray(size: Int): Array<EasyAdapterDataModel?> {
            return arrayOfNulls(size)
        }
    }
}